open-apparel-registry/open-apparel-registry
# Update search filters & enable autocomplete

## Overview

This PR updates the app to retrieve contributor, contributor type, and country options from the API, then to render them in auto-complete-enabled multiselect boxes in the sidebar search thing.

I stored the state of selections in Redux so they'll be available via `getState` for a facilities search request. Last commit here has basic facilities reducer + action setup which demonstrates that those values are available in the action (so they don't have to be read from the view). 

Connects #145 
Connects #81 

## Demo

![autocomplete](https://user-images.githubusercontent.com/4165523/51997445-bc99f480-2484-11e9-81d4-3330ad06d427.gif)

## Notes

I undertook a mild-to-medium refactor of the main page here: previously these search filters lived in a ControlPanel.jsx component that's pretty difficult to follow -- and that was embedded in the Map component.

I created a new FilterSidebar component for the search stuff so we can deprecate the ControlPanel, and I moved the search controls up one level in the component hierarchy so that they're now a sibling of the Map.jsx component. This required a mild change to the render method markup for Map.jsx.

I also removed two unused legacy reducers since they were no longer being used: Source and Navbar. This means that the Map reducer's the only one remaining that hasn't been redone with redux-act/immutability-helper.

## Testing Instructions

- serve this branch, then try out the sidebar search fields and verify that everything works as it should
- run `./scripts/test` and verify that everything passes
